File: targets/python-flask/routes/tools.py
"""Diagnostic tool routes."""
import os

from flask import Blueprint, request, jsonify

from services.tool_service import tool_service

tools_bp = Blueprint("tools", __name__, url_prefix="/tools")


@tools_bp.route("/ping")
def ping():
    host = request.args.get("host", "")
    rc = os.system("ping -c 1 " + host)
    return jsonify({"rc": rc})


@tools_bp.route("/diagnose")
def diagnose():
    host = request.args.get("host", "")
    tool_service.stage_target(host)
    rc = tool_service.run_staged_diag()
    return jsonify({"rc": rc})


@tools_bp.route("/calc", methods=["POST"])
def calc():
    expr = request.json.get("expr", "0")
    result = eval(expr)
    return jsonify({"result": result})

File: targets/js-ts-express/routes/tools.js
// Diagnostic tool routes.
const express = require('express');
const { exec } = require('child_process');
const { toolService } = require('../services/toolService');

const router = express.Router();

router.get('/ping', (req, res) => {
  const host = req.query.host;
  exec('ping -c 1 ' + host, (err, stdout) => {
    res.json({ out: stdout });
  });
});

router.get('/diagnose', (req, res) => {
  toolService.stageTarget(req.query.host);
  toolService.runStagedDiag((err, out) => {
    res.json({ out });
  });
});

router.post('/calc', (req, res) => {
  const expr = req.body.expr;
  const result = eval(expr);
  res.json({ result });
});

module.exports = router;
